modules/apps/ccv/parent/keeper/keeper.go

```diff
@@ -0,0 +1,256 @@
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.ChannelToChainKey(channelID), []byte(chainID))
+}
+
+// GetChannelToChain gets the baby chainID for a given CCV channelID
+func (k Keeper) GetChannelToChain(ctx sdk.Context, channelID string) (string, bool) {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.ChannelToChainKey(channelID))
+	if bz == nil {
+		return "", false
+	}
+	return string(bz), true
+}
+
+// SetUnbondingChanges sets the unbonding changes for a given baby chain and sequence
+func (k Keeper) SetUnbondingChanges(ctx sdk.Context, chainID string, seq uint64, valUpdates []abci.ValidatorUpdate) {
+	// TODO
+}
+
+// GetUnbondingChanges gets the unbonding changes for a given baby chain and sequence
+func (k Keeper) GetUnbondingChanges(ctx sdk.Context, chainID string, seq uint64) []abci.ValidatorUpdate {
+	// TODO
+	return nil
+}
+
+// DeleteUnbondingChanges deletes the unbonding changes for a given baby chain and sequence
+func (k Keeper) DeleteUnbondingChanges(ctx sdk.Context, chainID string, seq uint64) {
+	// TODO
+}
+
+// SetChannelStatus sets the status of a CCV channel with the given status
+func (k Keeper) SetChannelStatus(ctx sdk.Context, channelID string, status ccv.Status) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(ccv.ChannelStatusKey(channelID), []byte{byte(status)})
+}
+
+// GetChannelStatus gets the status of a CCV channel
+func (k Keeper) GetChannelStatus(ctx sdk.Context, channelID string) ccv.Status {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(ccv.ChannelStatusKey(channelID))
+	if bz == nil {
+		return ccv.UNINITIALIZED
+	}
+	return ccv.Status(bz[0])
+}
+
+// VerifyChildChain verifies that the chain trying to connect on the channel handshake
+// is the expected baby chain.
+func (k Keeper) VerifyChildChain(ctx sdk.Context, channelID string) error {
+	// Verify CCV channel is in Initialized state
+	status := k.GetChannelStatus(ctx, channelID)
+	if status != ccv.INITIALIZING {
+		return sdkerrors.Wrap(ccv.ErrInvalidStatus, "CCV channel status must be in Initializing state")
+	}
+	_, tmClient, err := k.getUnderlyingClient(ctx, channelID)
+	if err != nil {
+		return err
+	}
+	// TODO: Verify consensus state against initial validator set of the child chain
+
+	// Verify that there isn't already a CCV channel for the child chain
+	if prevChannel, ok := k.GetChainToChannel(ctx, tmClient.ChainId); ok {
+		return sdkerrors.Wrapf(ccv.ErrDuplicateChannel, "CCV channel with ID: %s already created for child chain %s", prevChannel, tmClient.ChainId)
+	}
+	return nil
+}
+
+// SetChildChain ensures that the child chain has not already been set by a different channel, and then sets the child chain mappings in keeper,
+// and set the channel status to validating.
+// If there is already a ccv channel between the parent and child chain then close the channel, so that another channel can be made.
+func (k Keeper) SetChildChain(ctx sdk.Context, channelID string) error {
```

Comment:
I'd recommend renaming this function since it does more than setting the child chain. I see why it is useful to wrap these checks before setting the child chain